bundler/lib/dependabot/bundler/file_updater/lockfile_updater.rb

```diff
@@ -188,6 +188,13 @@ def cache_vendored_gems(definition)
           end
         end
 
+        # This is not officially supported and may be removed without notice.
+        def __keep_on_prune?(spec_name)
+          if (specs = ::Bundler.settings[:persistent_gems_after_clean])
```

Comment:
Will this get set from the `Gemfile`?

Reply:
Yep, I've improved the tests to show this